modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM714GetAllDocumentationTestCase.java

```diff
@@ -98,161 +73,83 @@ public void setEnvironment() throws Exception {
-
-        HttpResponse verifyApiResponse = apiPublisher.getApi(apiName, apiProvider, apiVersion);
-        JSONObject verifyApiJsonObject = new JSONObject(verifyApiResponse.getData());
-        assertFalse(verifyApiJsonObject.getBoolean("error"), "Error in Verify API Response");
-
+        providerName = publisherContext.getContextTenant().getContextUser().getUserName();
+        apiEndPointUrl = backEndServerUrl.getWebAppURLHttp() + API_END_POINT_POSTFIX_URL;
+        apiRequest = new APIRequest(apiName, apiContext, new URL(apiEndPointUrl));
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setTier(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setProvider(providerName);
+        apiRequest.setDescription(API_DESCRIPTION);
+        HttpResponse createResponse = restAPIPublisher.addAPI(apiRequest);
+
+        assertEquals(createResponse.getResponseCode(), 201, "Error in API Creation");
+        apis.add(0, createResponse.getData());
         //publish API
-        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest(apiName, apiProvider,
-                APILifeCycleState.PUBLISHED);
-
-        HttpResponse statusUpdateResponse = apiPublisher.changeAPILifeCycleStatus(updateRequest);
-        assertEquals(statusUpdateResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Response Code is Mismatched");
-
-        JSONObject statusUpdateJsonObject = new JSONObject(statusUpdateResponse.getData());
-        assertFalse(statusUpdateJsonObject.getBoolean("error"), "API is not published");
-
-        docName = "TestDocumentInLine";
-        docType = "how to";
-        sourceTypeInLine = "inline";
-        String summary="TestFile";
-        String docLocation = "";
-
-        //add document with document type="how-to" and source type=inline;
-        HttpResponse addDocumentResponse = apiPublisher.addDocument(apiName, apiVersion, apiProvider, docName, docType,
-                sourceTypeInLine, docUrl, summary, docLocation);
-        assertEquals(addDocumentResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                "Error in Add Document Response Code");
-        JSONObject addDocumentJsonObject = new JSONObject(addDocumentResponse.getData());
-        assertFalse(addDocumentJsonObject.getBoolean("error"), "Response Error in Add Document: " + docName);
+        restAPIPublisher
+                .changeAPILifeCycleStatus(createResponse.getData(), APILifeCycleAction.PUBLISH.getAction(),
+                        null);
 
-        //add document with document type="how-to" and source type=url
-        sourceDocUrl = "https://docs.wso2.com/display/AM191/Tutorials";
-        sourceTypeUrl = "url";
-        docNameWithUrl = "TestDocumentWithUrl";
+        HttpResponse applicationResponse = restAPIStore.createApplication(applicationName,
```

Comment:
We won't be needing an application and subscription for this test

Reply:
Fixed